### ISSUE TITLE
Deprecated `pvsystem.sapm_spectral_loss` not removed

### ISSUE THREAD

**Issue by RDaxini** [pvlib/pvlib-python#2243]
**Describe the bug**
`pvsystem.sapm_spectral_loss` was deprecated since v10.0 but I could not find any PRs to remove it in a future version (12.0?)

**To Reproduce**
See the [read the docs page](https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.pvsystem.sapm_spectral_loss.html) 
and the [tests](https://github.com/pvlib/pvlib-python/blob/cbe4cc5a8bb3bb529f2dd885aebcd527fedbbc63/pvlib/tests/test_pvsystem.py#L236-L239)

**Expected behavior**
Removed in v11.0

**Versions:**
 - ``pvlib.__version__``: 11.2

**Additional context**
I am not super familiar with deprecations so please correct me if I have misunderstood something here or even missed an existing PR for this...

**Comment by kandersolar** [comment]
Yes, removing functions is done in `.0` releases, so this will wait until 0.12.0 at the earliest.  It's okay to leave #2244 open in the meantime, but opening PRs "early" like that means you may have to resolve some merge conflicts between now and then :)
